### ISSUE TITLE
Inconsistent User Entity classes

### ISSUE THREAD

**Issue by truckee** [symfony/symfony-docs#9955]
The User classes shown on pages [How to Load Security Users from the Database](https://symfony.com/doc/current/security/entity_provider.html) and [How to Implement a Simple Registration Form](http://symfony.com/doc/current/doctrine/registration_form.html) are inconsistent. For one, the Security Users page includes
````
    /**
     * @ORM\Column(name="is_active", type="boolean")
     */
    private $isActive;

    public function __construct()
    {
        $this->isActive = true;
        // may not be needed, see section on salt below
        // $this->salt = md5(uniqid('', true));
    }
````
whereas the Simple Registration page instead has 
````
    public function __construct()
    {
        $this->roles = array('ROLE_USER');
    }
````

One would expect them either to be the same or to include an explanation for why they are different.

**Comment by HeahDude** [comment]
Hello @truckee, thank you for opening this issue.

I would say that the docs has not been written in one day, nor with one application example in mind. It is years of efforts from the community.
So usually short code samples intend to demonstrate a use case of why we should do or need things in some particular context.
Therefor some examples are very specific to an idea and can bring inconsistency with other sections.
What we try to do in general is to keep consistency between examples for a same page, but keeping it consistent across all the docs is not that obvious, given also that all features are not meant to be used in a same simple application.

To conclude I would say that sometimes it is worth to bring some consistency between them, sometimes not.

I leave this issue open as a reminder that we should pass on these examples to trim them and show only what's needed in context. Thanks!

**Comment by javiereguiluz** [comment]
Thanks for reporting! However, I'm closing it as fixed because in #10423 we did a huge overhaul of Security docs and the `entity_provider.rst` article has been updated and we no longer show that class constructor. Thanks!
